rem/aufile/aufile.c

```diff
@@ -252,26 +252,26 @@ size_t aufile_get_size(struct aufile *af)
  *
  * @return length in ms if success, otherwise 0.
  */
-size_t aufile_get_length(struct aufile *af, struct aufile_prm *prm)
+size_t aufile_get_length(struct aufile *af, const struct aufile_prm *prm)
 {
-	if (!af)
+	if (!af || !prm)
 		return 0;
 
 	switch (prm->fmt) {
 		case AUFMT_PCMA:
 		case AUFMT_PCMU:
-			return af->datasize * prm->channels * prm->srate
-				/ 1000;
+			return af->datasize / prm->channels * 1000
+				/ prm->srate;
```

Comment:
Maybe more readable and only one division:
```c
return af->datasize * 1000 / (prm->channels * prm->srate);
```

Reply:
Should we also take the WAV header size into account?
See function `aufile_set_position()`!

Reply:
`af->datasize` is already the size of the data chunk without the header